### PR TITLE
added karaf feature definitions for addons

### PR DESCRIPTION
**Pull request by kaikreuzer**
Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

**Comment by kaikreuzer** [comment]
@maggu2810 This is the first step to move the feature definitions to the place where the code of the according bundles resides. I will do the very same for all openHAB 1 addons (define features in the openHAB 1 repo). The distribution will then be condensed to something like I started in https://github.com/kaikreuzer/openhab2/tree/distro

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,4 @@ distribution/openhabhome/userdata/logs/openhab.log
 distribution/openhabhome/userdata/mapdb/storage.mapdb
 distribution/openhabhome/userdata/mapdb/storage.mapdb.p
 distribution/openhabhome/userdata/mapdb/storage.mapdb.t
+features/openhab-addons/src/main/history
```
Comment (maggu2810) [inline .gitignore]: FYI: I am using <code>features/*/src/main/history</code> on my projects.

**Comment by maggu2810** [comment]
Looks good.
Not part of this PR themselves, but I would prefer (you have already done this in this PR) to use versioned (non-SNAPSHOT) dependencies for the OH2 core, and other non-addons stuff ... all the time. Just to ensure that no changes of the core will break the addon verification (a bump of the dependency in the addon could break it, but this is okay then).
All fine, feel free to merge (don't know if you would like to change the gitignore to use the wildcard).

**Comment by kaikreuzer** [comment]
Thanks for your review, I have updated the gitignore.

**Pull request merged**